### PR TITLE
Merge Smokeylab 37 into master.

### PR DESCRIPTION
**Pull request by danielmuhr**

### CODE REVIEW

File: app/emulator/include/emulator.hpp [hunk 1]
```diff
@@ -125,7 +126,7 @@ class EmulatorData {
   bool pindle_drive = false;
   bool pindle_reverse = false;
   bool parking_flag = false;
-  std::recursive_mutex emulator_data_mutex_;
+  std::shared_timed_mutex emulator_data_mutex_;
```
Comment (Dhax0r) [inline app/emulator/include/emulator.hpp]: maybe a std::shared_mutex instead?

File: app/emulator/src/emulator_data.cpp [hunk 2]
```diff
@@ -1,107 +1,107 @@
...
 }
 bool EmulatorData::GetParkingFlag() {
-  std::lock_guard<std::recursive_mutex> guard(emulator_data_mutex_);
+  std::shared_lock lock_s(emulator_data_mutex_);
   return parking_flag;
 }
 
 void EmulatorData::SetThrottleSetValue(const size_t& throttle_in_set_val) {
-  std::lock_guard<std::recursive_mutex> guard(emulator_data_mutex_);
+  std::unique_lock lock_u(emulator_data_mutex_, std::defer_lock);
```
Comment (Dhax0r) [inline app/emulator/src/emulator_data.cpp]: std::defer_lock would not acquire the lock, right?

**Review by Dhax0r — CHANGES_REQUESTED** [review]

File: app/emulator/src/emulator_data.cpp (hunk 2 above)
Comment (danielmuhr) [reply app/emulator/src/emulator_data.cpp]: True, I misunderstood. We need to talk about the implementation a bit more!

Comment (danielmuhr) [reply app/emulator/src/emulator_data.cpp]: Removed, let's talk about implementation.

**Pull request merged**